.github/workflows/main.yml

```diff
@@ -7,25 +7,36 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up JDK 1.8
-      uses: actions/setup-java@v1
-      with:
-        java-version: 1.8
-    - uses: derongan/nmsaction@v1
-      with:
-        rev: 1.15
-    - name: Install nms into m2
-      run: mkdir -p $HOME/.m2/repository && cp -a nms-build/.m2/repository/. $HOME/.m2/repository
-    - name: Grant execute permission for gradlew
-      run: chmod +x gradlew
-    - name: Build with Gradle
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: ./gradlew shadowJar
-    - name: Upload artifact
-      uses: actions/upload-artifact@v1.0.0
-      with:
-        name: package
-        # Directory containing files to upload
-        path: build/libs
\ No newline at end of file
+      - uses: actions/checkout@v2
+      - name: Load gradle cache
+        uses: actions/cache@v1
+        id: cache
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
```

Comment:
I'm not 100% sure how the restore keys work, but I think this will create a new cache whenever our gradle files change, and if no existing cache for it is found, it will try to get a previous cache using the restore keys. There's more info [here](https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key).

Reply:
yes thats correct. Good way of doing it